### DISCUSSION TITLE
Prebuilt Windows binaries with CLBlast, cuBlast, OpenBlas

### DISCUSSION THREAD

**Discussion by ilovefreesw** [ggml-org/whisper.cpp#1168] [Q&A]
Can you provide Windows binaries for Whisper.cpp with CLBlast, cuBlast, OpenBlas.... just like Llama.cpp provides it...?

![image](https://github.com/ggerganov/whisper.cpp/assets/77317006/1679d873-d652-4f57-96f3-8aa56321cbe3)

Right now, it's so exhausting and tedious work to build and compile this on Windows.

**Comment by bobqianic** [comment]
> Right now, it's so exhausting and tedious work to build and compile this on Windows.

That's not exactly true. Building these binaries on Windows using CMake is actually quite straightforward. Check this https://github.com/ggerganov/whisper.cpp/issues/878#issuecomment-1586192170

**Comment by ilovefreesw** [comment]
> > Right now, it's so exhausting and tedious work to build and compile this on Windows.

Installing VS Code where I live is a herculean task...due to slow internet. Downloading and setting up 5 Gb of packages just to compile this ... this is exactly what the issue is all about.

**Comment by magicse** [comment]
@ilovefreesw 
Try use Cmake and  mingw64 for building Windows base binaries.
MinGW very compact compiler.

**Comment by ap653998** [comment]
Fully agree. we just need binaries to play with like others provide already.

I was just tried to build on Windows without VS, using a GCC compiler.

I even had to tweak CMakeFile to be able to compile in the first place!
And not all the binaries are working or working properly, my suspicion.. some compiler flags forgotten or just different from one version to another (like -mavx2)

Instead of focusing my application, I had to spent 1 week to figure out what's wrong in the building process.
I end up taking binaries from other projects like Whisper.net and use from there :)

For you guys, which are guru's in whisper.cpp, probably it's a 30min effort to make a decent release package.

**Comment by timtak** [comment]
I am a 60 year old English Conversation teacher. The ability to provide an English conversation agent to my students would be a game changer, especially if I could tell it what to have them practice. I was thinking of providing ChatGPT to some of my students from April but that is like 20USD a month per student per month, so a installable thing would be brilliant. But building a working voice chattable model looks difficult. I hope there are installables one day not so far away. My students generally have Windows PCs.

**Comment by ilovefreesw** [comment]
I think this should be closed... the binaries are available in the CI: https://github.com/ggerganov/whisper.cpp/actions/runs/13173996633

@timtak @ap653998

**Reply by timtak** [reply]
Thank you. That looks brilliant. It would be even better if these downloads contained a readme. I tried clicking on "main.exe" but nothing happened.

**Reply by ilovefreesw** [reply]
Because it is a command line program. You run it from Terminal, PowerShell or CMD.exe. Commands are listed on the Readme.md of the project.

https://github.com/ggerganov/whisper.cpp/blob/master/README.md

**Reply by timtak** [reply]
Thank you. I think it is a bit too difficult for me.

**Reply by kgidwani** [reply]
@timtak, I am currently building an app for it.  I will update you once it is ready.
